@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -999,17 +1002,69 @@ export class TextElementsRenderer {
 
         const groupedPriorityLists: Map<number, TextElementLists> = new Map();
 
+        // Cache for point labels which may have duplicates in same tile or in neighboring tiles.
+        const pointLabelCache: Map<string, TextElement> = new Map();
+
+        // Determine which of the two labels were rendered last. This allows to keep the render
+        // state for fading.
+        const latestTextElement = (a: TextElement, b: TextElement): TextElement => {
+            if (a.textRenderState === undefined || a.textRenderState!.lastFrameNumber < 0) {
+                return b;
+            }
+            if (b.textRenderState === undefined || b.textRenderState!.lastFrameNumber < 0) {
+                return a;
+            }
+
+            return a.textRenderState!.lastFrameNumber > b.textRenderState!.lastFrameNumber ? a : b;
+        };
+
         for (const tile of tilesToRender) {
             for (const group of tile.textElementGroups.groups.values()) {
                 if (group.elements.length === 0) {
                     continue;
                 }
                 const foundGroup = groupedPriorityLists.get(group.priority);
                 if (foundGroup === undefined) {
+                    const uniqueLabels: TextElement[] = [];
+
```

Comment:
Do we need to clear the pountLabelCache here?

Reply:
The pointLabelCache should be applied over all tiles (outer loop), so we identify all duplicates at tile borders.

Reply:
Ok, then should `uniqueLabels`also live outside the loop? Can't the `indexOf` call fail if the cached TextElement is in another tile?